[WIP] 30 create input field (#35)
CrowdStrike/ember-toucan-core: docs/components/input/index.md

# Input 

> Provides a Toucan-styled [input element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input). If you are building forms, you may be interested in the TextareaField component instead.

Reviewer: ```suggestion
Provides a Toucan-styled [input element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input). If you are building forms, you may be interested in the InputField component instead.
```

> Provides a Toucan-styled [textarea element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/textarea). If you are building forms, you may be interested in the TextareaField component instead.

Reviewer 2: ```suggestion
Provides a Toucan-styled [input element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input). If you are building forms, you may be interested in the InputField component instead.
```

## Value

> To set the `value` attribute of the `<textarea>`, provide `@value`.

Reviewer 2: ```suggestion
To set the `value` attribute of the `<input>`, provide `@value`.
```

```hbs
<Form::Controls::Input @value='value' />
```

## onChange

To tie into the input event, provide `@onChange`. `@onChange` will return two arguments, the first being the value, while the second being the raw event.

```hbs
<Form::Controls::Input
  @value={{this.value}}
  @onChange={{this.handleChange}}
/>
```

```js
import Component from '@glimmer/component';
import { action } from '@ember/object';
import { tracked } from '@glimmer/tracking';
export default class extends Component {
  @tracked value;
  @action
  handleChange(value, e) {
    console.log({ e, value });
    this.value = value;
  }
}
```

## Disabled State

> Set the `@isDisabled` argument to disable the `<input>`.

Reviewer 2: Nit: May want to add info about `@hasError` :

```suggestion
Set the `@isDisabled` argument to disable the `<input>`.

## Error State

Set the `@hasError` argument to apply an error box shadow to the `<input>`.
```
